packages/expo-image/android/src/main/java/expo/modules/image/ImageLoadTask.kt

```diff
@@ -0,0 +1,34 @@
+package expo.modules.image
+
+import android.graphics.drawable.Drawable
+import com.bumptech.glide.Glide
+import expo.modules.kotlin.Promise
+import expo.modules.image.records.SourceMap
+import expo.modules.kotlin.AppContext
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.withContext
+
+open class ImageLoadTask(private val appContext: AppContext, private val source: SourceMap) {
+  private var task: Job? = null
+
+  suspend fun load(promise: Promise?) {
+    return coroutineScope {
+      val deferred = async {
+        val context = this@ImageLoadTask.appContext.reactContext ?: throw Exception("React context is not available")
```

Comment:
```suggestion
        val context = this@ImageLoadTask.appContext.reactContext ?: throw Exceptions.ReactContextLost()
```
You can use common exceptions to handle this case.